pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,243 @@
+from .._version import __version__
+
+try:
+    import dask
+    import dask.distributed as dist
+    from dask.distributed import worker_client, performance_report
+
+    from distributed.protocol import (
+        dask_serialize,
+        dask_deserialize,
+    )
+
+    @dask_serialize.register(qml.numpy.tensor)
+    def serialize(tensor: qml.numpy.tensor) -> Tuple[Dict, List[bytes]]:
+        "Defines a serializer for the Dask backend"
+        header, frames = dist.protocol.numpy.serialize_numpy_ndarray(tensor)
+        header["type"] = "qml.numpy.tensor"
+        header["requires_grad"] = tensor.requires_grad
+        frames = [tensor.data]
+        return header, frames
+
+    @dask_deserialize.register(qml.numpy.tensor)
+    def deserialize(header: Dict, frames: List[bytes]) -> qml.numpy.tensor:
+        "Defines a deserializer for the Dask backend"
+        return qml.numpy.tensor(frames[0], requires_grad=header["requires_grad"])
+
+
+except ImportError as e:  # pragma: no cover
+    raise ImportError("task.qubit requires installing dask and dask.distributed") from e
+
+
+class ProxyInstanceCLS(classmethod):
+    """
+    This utility class allows the use of both an instance
+    as well as class method types. For situations where
+    the explicit class method is needed, we can use this as a
+    decorator, and select the appropriate call dynamically.
+    This is an essential support for accessing backend
+    functionality supports with the proxy `task.qubit`.
+
+    The implementation is based on the Python descriptor guide
+    as mentioned https://stackoverflow.com/questions/28237955/same-name-for-classmethod-and-instancemethod/28238047#28238047
+    """
+
+    def __get__(self, instance, type_):
+        descr_get = super().__get__ if instance is None else self.__func__.__get__
+        return descr_get(instance, type_)
+
+
+# pylint: too-few-public-methods, bad-classmethod-argument
+class TaskQubit(QubitDevice):
+    """Proxy simulator plugin written using Dask.Distributed as a task-distribution scheduling backend.
+
+    **Short name:** ``task.qubit``
+
+    This device provides a pure-state qubit simulator wrapping both ``"default.qubit"`` and ``"lightning.qubit"``,
+    and written to allow batched offloading to a Dask scheduler.
+
+    To use this device, you will need to install dask and dask.distributed:
+
+    .. code-block:: console
+
+        pip install dask distributed
+
+    **Example**
+
+    >>> import pennylane as qml
+    >>> import tensorflow as tf
+    >>> import dask.distributed as dist
+
+    >>> if __name__ == '__main__': # Required for LocalCluster
+    ...     cluster = dist.LocalCluster(n_workers=4, threads_per_worker=1)
+    ...     client = dist.Client(cluster)
+    ...     backend = "default.qubit"
+    ...     dev = qml.device("task.qubit", wires=6, backend=backend)
+    ...     @qml.qnode(dev, cache=False, interface="tf", diff_method="parameter-shift) # caching must be disabled due to proxy interface
```

Comment:
I'm quite impressed this works with parameter-shift! If you change to backprop, does it still work for gradients?

Reply:
Yep, still works. The device internally expects to be calling functions from a dask worker (as in, you cannot explicitly run calculations on the device without submission to the Dask client). In this case, for param-shift, all subsequent tasks are spawned and scheduled to run on all available devices.
That being said, brackprop is obviously still going to be faster, but the option for both seemed to run just fine.

Reply:
```suggestion
    ...     backend = "default.qubit"
    ...     dev = qml.device("task.qubit", wires=6, backend=backend)
    ...     @qml.qnode(dev, cache=False, interface="tf", diff_method="parameter-shift") # caching must be disabled due to proxy interface
```